app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +107,57 @@ class SettingsActivity : AppCompatActivity() {
             .commit()
     }
 
+    override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        val menuItem = menu!!.findItem(R.id.app_bar_search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.queryHint = getString(R.string.search)
+
+        searchView.setOnQueryTextFocusChangeListener { _, focus ->
+            (binding.titlebar.toolbar.layoutParams as AppBarLayout.LayoutParams).scrollFlags =
+                if (focus)
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_NO_SCROLL
+                else
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+        }
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query : String) : Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText : String) : Boolean {
+                val queries = newText.split(",").toTypedArray()
+                if (newText.isNotEmpty()) {
+                    preferenceFragment.preferenceScreen.forEach { preferenceCategory ->
+                        val categoryVisible = queries.any { preferenceCategory.title?.contains(it, true) ?: false }
+                        // Tracks whether all preferences under this category are hidden
+                        var areAllPrefsHidden = true
+                        // Tracks whether there is at least one preference under this category that has a key
+                        var hasNonNullPreferenceKey = false
+                        (preferenceCategory as PreferenceCategory).forEach { preference ->
+                            preference.isVisible = categoryVisible || (preference.key != null && queries.any { preference.title?.contains(it, true) ?: false })
+                            if (preference.isVisible && areAllPrefsHidden)
+                                areAllPrefsHidden = false
+                            hasNonNullPreferenceKey = hasNonNullPreferenceKey || preference.key != null
+                        }
+                        // Hide PreferenceCategory if none of its preferences match the search and neither the category title
+                        preferenceCategory.isVisible = hasNonNullPreferenceKey && (!areAllPrefsHidden || categoryVisible)
+                    }
+                } else { // If user input is empty, show all preferences
+                    preferenceFragment.preferenceScreen.forEach { preferenceCategory ->
+                        preferenceCategory.isVisible = true
+                        (preferenceCategory as PreferenceCategory).forEach { preference ->
+                            preference.isVisible = true
+                        }
+                    }
+                }
```

Comment:
Drop this `else` block as the case is already handled by the rest of the code if you apply my suggestion above.